Gems/ScriptCanvas/SourceModificationScripts/UpdateSpawnableNodes.py

```diff
@@ -0,0 +1,30 @@
+import sys
+import shutil
+from pathlib import Path
+
+fullPath = sys.argv[1]
+
+path = Path(fullPath)
+fileName = path.stem
+pathToFile = path.parent.resolve()
+extension = path.suffix
+
+fileNameBackup = fileName + "_bckup"
+fullBackupPath = (path.parent).joinpath(fileNameBackup).with_suffix(extension)
+
+shutil.copy(fullPath, fullBackupPath)
+
+print("Backup saved to \'%s\'" % fullBackupPath)
+
+with open(fullPath, 'r') as file :
+  filedata = file.read()
+
+filedata = filedata.replace('"$type": "SpawnTicketInstance"', '"$type": "AzFramework::EntitySpawnTicket"')
+filedata = filedata.replace('"$type": "SpawnableAsset"', '"$type": "AzFramework::Scripts::SpawnableScriptAssetRef"')
+filedata = filedata.replace('\"Asset\"', '\"asset\"')
```

Comment:
This sounds risky because 'Asset' is such a common term that it might be used by other nodes in the SC file. Can you find a better replace string that only modifies the asset names within the Spawn nodes context?

Reply:
this one's tricky, because "Asset" is on an individual line. I'd have to some funky regex to do it right I think.

Reply:
Do you need a regex, or can you just parse the incoming json and detect that you actually have the key, value or pair that you are looking for? converting "Asset" to "asset" on any "Asset" in the file is likely to risky for modifying source data.

Reply:
json looks like this:
![image](https://user-images.githubusercontent.com/82239319/168339973-eb15cfa4-c232-4ceb-a9e2-9a4b28d654fc.png)
I could maybe change it to `"asset": {`

Reply:
alternative is adding custom serializer to SpawnableAssetRef

Reply:
maybe there's some python json parser I could also use?

Reply:
Yeah you can parse the actual json from python, which will be much safer: https://docs.python.org/3/library/json.html
```
import json

prefab_dom = json.loads(filedata)
# inspect DOM, do fixup - it'll be stored as a python dict
filedata = json.dumps(prefab_dom, indent=4)
```